app/stac_api/serializers.py

```diff
@@ -338,13 +350,61 @@ def validate_type(self, value):
         '''
         return normalize_and_validate_media_type(value)
 
+    def _validate_href_url(self, url):
+        # thanks https://stackoverflow.com/questions/3809401/
+        # what-is-a-good-regular-expression-to-match-a-url
+        url_regex = (
+            r'https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}'
+            r'\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)'
+        )
+
+        if not re.match(url_regex, url):
+            errors = {'href': _('Invalid URL provided')}
+            raise serializers.ValidationError(code='payload', detail=errors)
+
+        # get the collection regex. if there's none, allow all urls
+        collection_regex = self.collection.external_asset_pattern or r".*"
+
+        if not re.search(collection_regex, url):
+            errors = {'href': _("Invalid URL provided. It doesn't match the collection pattern")}
+            raise serializers.ValidationError(code='payload', detail=errors)
+
```

Comment:
Regarding also @benschs comment: It would probably be safer to have prefixes and check with `startswith`, instead of just searching in the whole string.